app/services/service_provider_request_handler.rb

```diff
@@ -7,6 +7,8 @@ def initialize(url:, session:, protocol_request:, protocol:)
   end
 
   def call
+    return if sp_url_stored_in_session == url
```

Comment:
This has the same vibes as the change [here](https://github.com/18F/identity-idp/pull/2563/files#diff-7fcf9b31a11b5c54dd038dc3e6afb2b883bff6c1579a9460776e31bee5ba090cR10) that was fixed [here](https://github.com/18F/identity-idp/pull/3832/files#diff-7fcf9b31a11b5c54dd038dc3e6afb2b883bff6c1579a9460776e31bee5ba090cL10). This should still allow for the same service provider with differing nonces (or other attributes).